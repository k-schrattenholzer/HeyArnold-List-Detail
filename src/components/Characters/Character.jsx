export default function Character({ character }) {
  return (
    <section>
      <h1>{character.name}</h1>
      <img src={character.image} alt={character.name} />
      <p>id:{character._id}</p>
    </section>
  )
}