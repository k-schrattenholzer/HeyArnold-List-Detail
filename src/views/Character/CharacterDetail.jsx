import { useState } from "react";
import { useParams } from "react-router-dom"

export default function CharacterDetail() {
  const { id } = useParams();
  const [loading, setLoading] = useState(true);
  const [character, setCharacter] = useState({});

  return (
    <div>
      This is the detail page for a character with ID of {id}
    </div>
  )
}
