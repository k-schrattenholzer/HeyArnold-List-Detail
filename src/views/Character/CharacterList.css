@import '../../App.css';

li {
  list-style-type: none;
}

img {
  width: 40px;
  height: 50px;
}

.characterListContainer {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 1fr 1fr;
}

.characterListEl {
  grid-column: span 2;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.imageContainer {
  width: auto;
}
.imageEl {
  opacity: 1;
  display: block;
  width: 200px;
  height: auto;
  transition: .5s ease;
  backface-visibility: hidden;
  margin: 10px;
  border-radius: 85px;
}

.imageContainer:hover .imageEl {
  opacity: 0.3;
}
