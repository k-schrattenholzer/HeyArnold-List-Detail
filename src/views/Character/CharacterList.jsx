import { useEffect, useState } from "react"
import { Link } from "react-router-dom";
import Characters from "../../components/Characters/Characters.jsx";
import { getCharacters } from "../../services/heyArnoldApi.js"
import styles from '../../App.css'

export default function CharacterList() {
  const [loading, setLoading] = useState(true);
  const [characterList, setCharacters] = useState([]);

  useEffect(async () => {
    try {
      const res = await getCharacters();
      setCharacters(res);
      setLoading(false);
    } catch (error) {
      console.log(error.message)
    }
  }, [])

  return (
    <div className={styles.characterListContainer}>

      <div>
        <h1>this is the character list</h1>
        <p>click on a character to get more information</p>
        <Link to='/'>home</Link>
      </div>

      {loading ? <p>loading up</p> : <Characters characterList={characterList}/>
      }

      
    </div>
  )
}
