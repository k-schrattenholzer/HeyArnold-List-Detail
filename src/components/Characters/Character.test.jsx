import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Character from './Character.jsx';

it('should render the characters component', () => {
  const testCharacter =
    {
      _id: 1,
      name: 'Grandma Gertie',
      image: 'https://website.com/image/gertie.jpg',
    }

  const { container } = render (
    <MemoryRouter>
      <Character character={testCharacter}/>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});