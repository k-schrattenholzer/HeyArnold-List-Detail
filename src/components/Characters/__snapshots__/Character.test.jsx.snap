// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the characters component 1`] = `
<div>
  <section>
    <h1>
      Grandma Gertie
    </h1>
    <img
      alt="Grandma Gertie"
      src="https://website.com/image/gertie.jpg"
    />
    <p>
      id:
      1
    </p>
  </section>
</div>
`;
